cpp/src/parquet/metadata.cc

```diff
@@ -311,15 +312,11 @@ class ColumnChunkMetaData::ColumnChunkMetaDataImpl {
         descr_->sort_order() == SortOrder::UNKNOWN) {
       return false;
     }
-    {
-      // Because we are modifying possible_stats_ in a const method
-      const std::lock_guard<std::mutex> guard(stats_mutex_);
-      if (possible_stats_ == nullptr) {
-        possible_stats_ = MakeColumnStats(*column_metadata_, descr_);
-      }
+    if (possible_encoded_stats_ == nullptr) {
+      possible_encoded_stats_ =
+          std::make_shared<EncodedStatistics>(FromThrift(column_metadata_->statistics));
     }
-    EncodedStatistics encodedStatistics = possible_stats_->Encode();
```

Comment:
I agree that it is unnecessary to call `Encode()` every time `is_stats_set()`. Should we take a different approach by caching the result of `is_stats_set()`?

IMO, many Parquet metadata objects are meant to access once and cache them for follow-up uses. For example, `FileMetaData` will always create new `RowGroupMetaData` and `ColumnMetaData` objects so users are supposed to cache them to avoid creation cost.